terraform/monitoring/06_log_based_metric.tf

```diff
@@ -0,0 +1,83 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Creates a log-based metric by extracting a specific log written by the Checkout Service. 
+# The log being used for the metric is from the Checkout Service and has the format:
+# [CheckoutService] orderedItem="Vintage Typewriter"
+#
+# The label and Regex extractor are used to create filters on the metric
+# This resource creates only the metric
+resource "google_logging_metric" "checkoutservice_logging_metric" {
+  name   = "checkoutservice_log_metric"
+  filter = "resource.type=k8s_container AND resource.labels.cluster_name=cloud-ops-sandbox AND resource.labels.namespace_name=default AND resource.labels.container_name=server AND [CheckoutService]"
+  metric_descriptor {
+    metric_kind = "DELTA"
+    value_type  = "INT64"  # specifies our metric to be a counter-based metric
```

Comment:
Is it actually the value_type that defines this metric to be counter-based, not the metric_kind? Couldn't you have a double counter metric?

Reply:
I guess in theory you could have a double counter metric but according to the docs a counter based metric is specified by the INT64 value type. I guess the same goes for DELTA so I added a line saying why we chose those two values.